### PR TITLE
Collagegen

### PR DESCRIPTION
**Pull request by advpetc**

### CODE REVIEW

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 1]
```diff
@@ -1,22 +1,109 @@
...
-	
-	//download collage to the server
-	public void downloadCollage(Collage collage) {
-		try {
-			BufferedImage image = collage.collage; //getter?
-			String fileName = collage.topic;
-			fileName+= “_” + collage.id  + “.png”;    //if collage has a unique id?
-			File outputFile = new File(topic);
-			ImageIO.write(image, “png”, outputFile);
+	private ArrayList<String> imageUrls; //change to <BufferedImage> if necessary
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: BufferedImage

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 2]
```diff
@@ -1,22 +1,109 @@
...
+
+	public CollageGenerator(ArrayList<String> urls) {
+		this.imageUrls = new ArrayList<String>();
+		this.images = new ArrayList<BufferedImage>();
+		this.borderedImages = new ArrayList<BufferedImage>();
+		this.collage = new BufferedImage();
+
+		this.imageUrls = urls;
+		for(int i=0; i < imageUrls.size(); i++) {
+			URL url = new URL(imageUrls.get(i));
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: BufferedImage

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 3]
```diff
@@ -1,22 +1,109 @@
...
+	private ArrayList<String> imageUrls; //change to <BufferedImage> if necessary
+	private ArrayList<BufferedImage> images;
+	private ArrayList<BufferedImage> borderedImages;
+	private BufferedImage collage;
+
+	public CollageGenerator(ArrayList<String> urls) {
+		this.imageUrls = new ArrayList<String>();
+		this.images = new ArrayList<BufferedImage>();
+		this.borderedImages = new ArrayList<BufferedImage>();
+		this.collage = new BufferedImage();
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: There is no default constructor for BufferedImage initialization--requires params.

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 4]
```diff
@@ -1,22 +1,109 @@
 package csci310group11.Implementation;
 
-import javq.io.*;
-import java.awt.image.*;
-import javax.imageio.*;
+import java.awt.Graphics2D;
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: need to import java.net.URL if you are using URL

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 5]
```diff
@@ -1,22 +1,109 @@
...
+	 * Should we set collage size 1st or make collage based of images entered
+	 */
+	private void resizeImages() {
+		//1/20th of collage dimensions
+		int resizeWidth = this.collage.getWidth()/20;
+		int resizeHeight = this.collage.getHeight()/20;
+		
+		//Iterate through all images
+		for(int i=0; i < images.size(); i++) {
+			BuffereImage img = images.get(i);
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: typo for BufferedImage

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 6]
```diff
@@ -1,22 +1,109 @@
...
+			//New BufferedImage with 1/20th dimensions of collage
+			BufferedImage resizeImg = new BufferedImage(resizeWidth, resizeHeight, img.getType());
+
+			//Draws the img image into the size of the resizeImg
+			Graphics2D graphics = resizeImg.getGraphics();
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: need casting for resizeImg.getGraphics();

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 7]
```diff
@@ -1,22 +1,109 @@
...
+	/**
+	 * Responsible for adding a 3px white border to each image to be added to the collage.
+	 * 
+	 * Creates a new BufferedImage that is 6px taller and wider than the original BufferedImage in images.
+	 * 
+	 * Sets thes the graphics of the larger BufferedImage to white. Paints the original image onto the 
+	 * new BufferedImage to create a 3px "border". Adds the bordred BuffereImage to this.borderedImages.
+	 */
+	private void addBorderToImages() {
+		for(int i=0; i < image.size(); i++) {
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: typo for images

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 8]
```diff
@@ -1,22 +1,109 @@
...
+			BufferedImage image = image.get(i);
+			int width = image.getWidth();
+			int height = image.getHeight();
+			
+			//Create image with enough space for 3px border
+			BufferedImage borderedImage = new BufferedImage(width + 6, height + 6, image.getType());
+
+			//Setting larger image to all white
+			Graphics2D graphics = borderedImage.getGraphics();
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: need casting for borderedImage.getGraphics();

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 9]
```diff
@@ -1,22 +1,109 @@
...
+			graphics.setPaint(new Color(0,0,0));
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java]: import java.awt.Color

**Review by advpetc — COMMENTED** [review]
if you happened to use any constants, please use the constants from Constants class

File: src/main/java/csci310group11/Implementation/CollageGenerator.java (hunk 3 above)
Comment (allenyhu) [reply src/main/java/csci310group11/Implementation/CollageGenerator.java]: Yeah. We'll decide on that at the meeting.

File: src/main/java/csci310group11/Implementation/CollageGenerator.java (hunk 6 above)
Comment (allenyhu) [reply src/main/java/csci310group11/Implementation/CollageGenerator.java]: what casting is required?

Comment (advpetc) [reply src/main/java/csci310group11/Implementation/CollageGenerator.java]: (Graphics2D)

File: src/main/java/csci310group11/Implementation/CollageGenerator.java [hunk 10]
```diff
@@ -1,17 +1,73 @@
...
 
 public class CollageGenerator {
+	private ArrayList<BufferedImage> images; //change to <BufferedImage> if necessary
+	private ArrayList<BufferedImage> borderedImages;
+	private BufferedImage collage;
+
+	public CollageGenerator(ArrayList<BufferedImage> images) {
+		this.images = new ArrayList<BufferedImage>();
+		this.borderedImages = new ArrayList<BufferedImage>();
+		this.collage = new BufferedImage();
```
Comment (advpetc) [inline src/main/java/csci310group11/Implementation/CollageGenerator.java:20]: uhh i think it won't be able to initialize unless you initialize the size of the collage.

**Review by nihar-agrawal — APPROVED** [review]

**Pull request merged**